### DISCUSSION TITLE
How to connect to the spike hub through the browser on Linux

### DISCUSSION THREAD

**Discussion by TCE-1** [gpdaniels/spike-prime#48] [General]
Hello, we have a spike hub that we code through the Lego site with the block code for kids. We'd like to do this on out Linux (Ubuntu) laptop but haven't had any luck. It says Firefox isn't supported.

Does your project provide a GUI to do block coding or is this a tool strictly for written code?

If it is strictly for written code where would you suggest to look to learn more about how to code different Lego devices (motors, sensors, lights...)?

Thank you

**Comment by dlech** [comment]
You can use Chromium or Google Chrome on Linux.

You might also be interested in https://pybricks.com

**Reply by TCE-1** [reply]
Tried Chromium , didn't work. Will try Chrome next. Haven't heard of py bricks. Will check it out.
Thank you

**Reply by dlech** [reply]
In both browsers on Linux, you have to enable experimental features. Copy and paste this into the address bar to navigate to the place to turn it on.

    chrome://flags/#enable-experimental-web-platform-features

**Comment by gpdaniels** [comment]
This project is just me hacking around in python. If you want block coding I'd probably recommend you either stick with the official app from Lego, or try Pybricks.

One nice feature of these hubs is that they are very customisable and can be programmed just using a serial connection if needs be.

**Reply by TCE-1** [reply]
Thanks for the response. Do you have any recommendations on sources to read up on the potential / capabilities of the hubs?

**Reply by gpdaniels** [reply]
All the info I have is in this repo :smile:
